apis/user/v1alpha1/user_types.go

```diff
@@ -27,7 +27,19 @@ import (
 
 // UserParameters are the configurable fields of a User.
 type UserParameters struct {
+	// Group for the new user.
+	// +optional
+	// +immutable
 	GroupID string `json:"groupId"`
+
+	// GroupIDRef is a reference to a group to retrieve its groupId.
+	// +optional
+	// +immutable
+	GroupIDRef *xpv1.Reference `json:"groupIdRef,omitempty"`
+
+	// GroupIDSelector selects reference to a group to retrieve its groupId.
+	// +optional
+	GroupIDSelector *xpv1.Selector `json:"groupIdSelector,omitempty"`
```

Comment:
Is a name like `GroupIdRef` commonly used? I would except just `GroupRef`. As a user, I am referring to a group that I also have, and don't really care about what withinit makes it identifyable.

Reply:
Good point! Let me check,

Reply:
Seems like we can/should remove the ID term here from all fields.

Reply:
So `Group`, `GroupRef` and `GroupSelector`. I'll update the PR. Good catch!

Reply:
Done